internal/mode/static/state/graph/httproute.go

```diff
@@ -760,6 +762,44 @@ func validateFilterRedirect(
 	return allErrs
 }
 
+func validateFilterRewrite(
+	validator validation.HTTPFieldsValidator,
+	filter v1.HTTPRouteFilter,
+	filterPath *field.Path,
+) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if filter.URLRewrite == nil {
+		panicForBrokenWebhookAssumption(errors.New("urlRewrite cannot be nil"))
+	}
+
+	rewrite := filter.URLRewrite
+	rewritePath := filterPath.Child("urlRewrite")
+
+	if rewrite.Hostname != nil {
+		if err := validator.ValidateHostname(string(*rewrite.Hostname)); err != nil {
+			valErr := field.Invalid(rewritePath.Child("hostname"), *rewrite.Hostname, err.Error())
+			allErrs = append(allErrs, valErr)
+		}
+	}
+
+	if rewrite.Path != nil {
```

Comment:
Should we also be checking the path type here? The spec says:
> Type defines the type of path modifier. 
>
> Additional types may be added in a future release of the API.
Note that values may be added to this enum, implementations must ensure that unknown values will not cause a crash.
>
>Unknown values here must result in the implementation setting the Accepted Condition for the Route to status: False, with a Reason of UnsupportedValue.